core/src/test/java/org/apache/iceberg/TestMetadataUpdateParser.java

```diff
@@ -350,9 +485,51 @@ private static void assertEqualsAddSortOrder(
           UnboundSortOrder.UnboundSortField actualField = actual.sortOrder().fields().get(i);
           Assert.assertTrue("Fields of the sort order should be the same",
               expectedField.sourceId() == actualField.sourceId() &&
-              expectedField.nullOrder().equals(actualField.nullOrder()) &&
-              expectedField.direction().equals(actualField.direction()) &&
-              Objects.equals(expectedField.transformAsString(), actualField.transformAsString()));
+                  expectedField.nullOrder().equals(actualField.nullOrder()) &&
+                  expectedField.direction().equals(actualField.direction()) &&
+                  Objects.equals(expectedField.transformAsString(), actualField.transformAsString()));
         });
   }
+
+  private static void assertEqualsSetDefaultSortOrder(
+      MetadataUpdate.SetDefaultSortOrder expected, MetadataUpdate.SetDefaultSortOrder actual) {
+    Assert.assertEquals("Sort order id should be the same", expected.sortOrderId(), actual.sortOrderId());
+  }
+
+  // TODO - Come back to this when FileIO is injected. If this class extends TableTestBase, the validateSnapshot
+  //  methods can be reused. Also, this needs to handle V1 vs V2 for assertions. Or casting to BaseSnapshot,
+  //  BaseSnapshot::equals can be used.
+  private static void assertEqualsAddSnapshot(
+      MetadataUpdate.AddSnapshot expected, MetadataUpdate.AddSnapshot actual) {
+    Assert.fail("MetadataUpdate equality checking for AddSnapshot is not implemented yet");
+  }
+
+  private static void assertEqualsRemoveSnapshots(
+      MetadataUpdate.RemoveSnapshot expected, MetadataUpdate.RemoveSnapshot actual) {
+    Assert.assertEquals("Snapshots to remove should be the same", expected.snapshotId(), actual.snapshotId());
+  }
+
+  // See TODOs in MetadataUpdate.SetSnapshotRef.
+  private static void assertEqualsSetSnapshotRef(
+      MetadataUpdate.SetSnapshotRef expected, MetadataUpdate.SetSnapshotRef actual) {
+    Assert.fail("MetadataUpdate equality checking for SetSnapshotRef is not implemented yet");
+  }
```

Comment:
See the notes in `MetadataUpdate.SetSnapshotRef` about possible things to update.

Happy to remove this from this PR and place it elsewhere.

Reply:
I don't see the note you're referring to. The parser looks like it supports this, so I think it can be implemented. This shouldn't be affected by whether or not the optional fields are serialized using `null` or are omitted. Though you will want tests for reading all cases (omitted, null, non-null) and make sure they match here.

Reply:
I have added all of those tests. I basically copied the tests from `TestSnapshotRefParser` and updated them for this, with versions with versions that read all cases.